src/backend/aspen/api/views/samples.py

```diff
@@ -200,7 +208,9 @@ async def update_samples(
         # workaround for our response serializer
         sample.show_private_identifier = True
 
-        sample.generate_public_identifier(already_exists=True)
+        sample.generate_public_identifier(
+            pathogen_repo_config.prefix, already_exists=True
```

Comment:
We're passing the prefix into the name generator here so we're not stuck with `hcov-19/` for everything